cartoframes/io/carto.py

```diff
@@ -129,6 +129,9 @@ def to_carto(dataframe, table_name, credentials=None, if_exists='fail', geom_col
     elif has_geometry(dataframe):
         gdf.set_geometry(dataframe.geometry.name, inplace=True)
 
+        if GEOM_COLUMN_NAME in gdf and dataframe.geometry.name != GEOM_COLUMN_NAME:
```

Comment:
This code must be moved right before the rename_geometry because otherwise, every time the user passes a `geom_col` and the df contains a column `the_geom` it fails.